python/cudf/cudf/core/column/decimal.py

```diff
@@ -0,0 +1,69 @@
+import more_itertools
+import pyarrow as pa
+
+from cudf import _lib as libcudf
+from cudf.core.buffer import Buffer
+from cudf.core.column import ColumnBase
+from cudf.core.dtypes import DecimalDtype
+from cudf.utils.utils import pa_mask_buffer_to_mask
+
+
+class DecimalColumn(ColumnBase):
+    @classmethod
+    def from_arrow(cls, data: pa.Array):
+        mask_buf = data.buffers()[0]
+        mask = (
+            mask_buf
+            if mask_buf is None
+            else pa_mask_buffer_to_mask(mask_buf, len(data))
+        )
+        bts = data.buffers()[1].to_pybytes()
+        bts = b"".join(list(more_itertools.sliced(bts, 8))[::2])
+        return cls(
+            data=Buffer.from_bytes(bts),
+            size=len(data),
+            dtype=DecimalDtype.from_arrow(data.type),
+            mask=mask,
+        )
+
+    def to_arrow(self):
+        data_buf_64 = self.base_data.to_host_array()
+        zeros_buf = bytes(data_buf_64.size)
+        data_buf_128 = pa.py_buffer(
+            bytes(
+                more_itertools.flatten(
+                    more_itertools.interleave(
+                        more_itertools.chunked(data_buf_64, 8),
+                        more_itertools.chunked(zeros_buf, 8),
```

Comment:
I know @shwina made a comment about using numba to replace `more_itertools`, but unless I am missing a usage it seems like all the applications of this package are host side. In that case, I think you'd be O.K. just writing a normal loop in python to re-interleave the data with the zeros. My guess is you'd actually see similar performance. 

If you must use the GPU, could a combination of viewing the data as `int8`, `cupy` and [this answer](https://stackoverflow.com/questions/5347065/interweaving-two-numpy-arrays) do the job?

Reply:
Ideally, we want this to be on the GPU. You're 100% right about CuPy - that might end up being a much better solution than a custom Numba kernel.

Reply:
+1 to using cupy and avoiding Numba

Reply:
Fixed.